lib/src/ui/power_off_list/widgets/size_measurment_widget.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+typedef void OnWidgetSizeChange(Size size);
+
+class MeasureSizeRenderObject extends RenderProxyBox {
+  Size? oldSize;
+  OnWidgetSizeChange? onChange;
+
+  MeasureSizeRenderObject(this.onChange);
+
+  @override
+  void performLayout() {
+    super.performLayout();
+    Size newSize;
+    if (child != null) {
+      newSize = child!.size;
+    } else {
+      newSize = Size.zero;
+    }
+    if (oldSize == newSize) return;
+
+    oldSize = newSize;
+    WidgetsBinding.instance!.addPostFrameCallback((_) {
+      onChange!(newSize);
+    });
+  }
+}
+
+class MeasureSize extends SingleChildRenderObjectWidget {
+  final OnWidgetSizeChange onChange;
```

Comment:
I'd use `ValueChanged<T>` callback typealias from Flutter SDK. You can leave it as it is